health-checker/src/main/java/io/stargate/health/CheckerResource.java

```diff
@@ -47,7 +50,8 @@ public CheckerResource() {
   public Response checkLiveness() {
     // Sufficient to just return a 200 OK since liveness just means "app is up and doesn't need
     // restarted"
-    if (bundleService.checkBundleStates()) {
+    HealthCheck bundleHealthCheck = healthCheckRegistry.getHealthCheck(BUNDLES_CHECK_NAME);
+    if (bundleHealthCheck != null && bundleHealthCheck.execute().isHealthy()) {
```

Comment:
in what situation the `bundleHealthCheck == null`? Isn't it abnormal? Right now, if this happens to be null, we will return the status `UP`. It may be misleading.

Reply:
Not sure I follow, sorry. If `bundleHealthCheck` is `null` the execution will proceed to line 58 and return `SERVICE_UNAVAILABLE` :thinking: 

The check for `null` is just a basic precaution if case the code in other classes is changed and the check is not registered for some reason.

Reply:
You are right, sorry I missed that 👍